frontend/src/features/crawl-workflows/workflow-filter.ts

```diff
@@ -0,0 +1,81 @@
+import { localized } from "@lit/localize";
+import type { SlDropdown } from "@shoelace-style/shoelace";
+import { html } from "lit";
+import { customElement, property, query } from "lit/decorators.js";
+import { ifDefined } from "lit/directives/if-defined.js";
+
+import { TailwindElement } from "@/classes/TailwindElement";
+
+/**
+ * @slot
+ * @slot dropdown-header
+ * @slot dropdown-content
+ */
+@customElement("btrix-workflow-filter")
+@localized()
+export class WorkflowFilter extends TailwindElement {
+  @property({ type: Boolean })
+  checked?: boolean;
+
+  @property({ type: Boolean })
+  select?: boolean;
+
+  @property({ type: Boolean })
+  stayOpenOnChange?: boolean;
+
+  @query("sl-dropdown")
+  private readonly dropdown?: SlDropdown | null;
+
+  render() {
+    if (this.select) {
+      return html`
+        <sl-dropdown
+          distance="4"
+          hoist
+          ?stayOpenOnSelect=${this.stayOpenOnChange}
```

Comment:
It'd be nice to style the buttons when the dropdown is open (pt 1)
```suggestion
          ?stayOpenOnSelect=${this.stayOpenOnChange}
          class="group/dropdown"
```